### DISCUSSION TITLE
Did an entity come from the EntityRuler or the EntityRecognizer?

### DISCUSSION THREAD

**Discussion by clcarver1130** [explosion/spaCy#8036] [Help: Best practices]
Hi!

I have a custom-built pipeline that includes both an EntityRuler and a trained EntityRecognizer. Curious if there's a simple way to tell what piece of the pipeline (Ruler or Recognizer) an entity was generated by. I would like to compare the performance of both but can't separate them programmatically during analysis.

**Comment by polm** [comment]
There's no explicit attribute that tracks provenance of entities, and I'm not sure there's any implicit way you can tell them apart. 

I think typically you would compare the results of running each of them individually, or the result of running one component without the other, rather than tracking the provenance of entities. 

If you do want to see where each entity came from, you could add a separate component in between the two components. It could set token or span attributes on labels that exist when it runs, and you could use that to differentiate the output of the two components.
